ios/brave-ios/Sources/Brave/Frontend/UserContent/UserScripts/Scripts_Dynamic/Scripts/DomainSpecific/Paged/BraveSearchResultAdScript.js

```diff
@@ -60,5 +74,7 @@ window.__firefox__.includeOnce('BraveSearchResultAdScript', function($) {
     return jsonLdCreatives;
   };
 
-  sendMessage(getJsonLdCreatives());
+  setTimeout(() => {
+    sendMessage(getJsonLdCreatives());
+  }, 300);
```

Comment:
300 seconds seems rather random... How and why did we get to this number :-)

Reply:
Discussed further with @aseren and this is a plaster due to ads being shown for a short period of time. These kinds of plasters will always fall off and come back to bite us. Lets wait to hear from the iOS team and come up with a robust solution. Thank

Reply:
Basing on discussion there is no reliable and convenient way for detection if ad is blocked by ad-blocker. I'm declining the PR for now, until the proper solution is found.